frontend/src/layouts/EditPublishLayout.tsx

```diff
@@ -33,21 +41,47 @@ const EditPublishLayout: React.FC<EditPublishLayoutProps> = ({
             onChange={(e) => setTitle(e.target.value)}
           />
         </div>
-        {/* <TextEdtior content={content} setContent={setContent} /> */}
         <ReactQuill
           theme="snow"
           value={content}
           onChange={setContent}
         ></ReactQuill>
         <button
           type="submit"
-          onClick={() => submitFunction({ title, content })}
-          className="w-[150px] items-center px-5 py-2.5 text-sm font-medium text-center text-white bg-blue-700 rounded-lg focus:ring-4 focus:ring-blue-200 hover:bg-blue-800"
+          onClick={handleSubmit}
+          className="w-[150px] flex justify-center items-center px-5 py-2.5 text-sm font-medium text-center text-white bg-blue-700 rounded-lg focus:ring-4 focus:ring-blue-200 hover:bg-blue-800 disabled:opacity-50 disabled:cursor-not-allowed"
+          disabled={isLoading}
         >
-          {submitFunctionName}
+          {isLoading ? (
+            <>
+              <svg
+                className="animate-spin mr-2 h-5 w-5 text-white"
+                xmlns="http://www.w3.org/2000/svg"
+                fill="none"
+                viewBox="0 0 24 24"
+              >
+                <circle
+                  className="opacity-25"
+                  cx="12"
+                  cy="12"
+                  r="10"
+                  stroke="currentColor"
+                  strokeWidth="4"
+                ></circle>
+                <path
+                  className="opacity-75"
+                  fill="currentColor"
+                  d="M4 12a8 8 0 018-8v4a4 4 0 00-4 4H4z"
+                ></path>
+              </svg>
+              Please wait
+            </>
```

Comment:
The app already has a spinner component. Can we reuse it instead of creating an additional spinner svg here ?